docs/prompt_engineering/tutorials/best_practices.mdx

```diff
@@ -0,0 +1,134 @@
+  - **The Check:** The pipeline then compares the new prompt's aggregate evaluation scores (e.g., average correctness, helpfulness score, latency, cost) against the current production prompt's scores.
+- **Stage 4: Continuous deployment (promotion)**
+  Based on the evaluation results, the prompt is automatically promoted.
+  - **Pass/Fail Logic:** The pipeline checks if the new prompt is "better" than the current one (e.g., higher correctness score, no drop in helpfulness, within cost budget).
+  - **Promotion to `staging`:** If it passes, a script uses the LangSmith SDK to move the `staging` tag to this new commit hash. Your staging application, which pulls the `your-prompt:staging` tag, will automatically start using the new prompt.
+  - **Promotion to `prod`:** This is often a **manual step**. After the team has reviewed the performance in the staging environment and is confident in the results, a team member can manually move the `prod` tag in the LangSmith UI. This final human check is a crucial safety measure.
+
+## Syncing prompts in production
+
+For better version control, collaboration, and integration with your existing CI/CD pipelines, synchronizing your LangSmith prompts with an external source code repository will give you a full commit history alongside your application code.
+
+### Best practice: use webhooks for synchronization
+
+The most effective way to automate this is by using webhooks. You can configure LangSmith to send a notification to a service every time a new version of a prompt is saved. This creates a seamless bridge between the user-friendly prompt editing environment in LangSmith and your rigorous version control system.
+
+### Webhook synchronization flow
+
+This diagram shows the sequence of events when a prompt is updated in LangSmith and automatically synced to a GitHub repository.
+
+![CICD Diagram](./static/prompt-best-practice-sequence.png)
+
+### How to implement this with LangSmith
+
+LangSmith allows you to configure a webhook for your workspace that will fire on every prompt commit. You can point this webhook to your own service (like an AWS Lambda function or a small server) to handle the synchronization logic.
+
+**➡️ Learn more in the official documentation:** [Trigger a webhook on prompt commit](https://docs.smith.langchain.com/prompt_engineering/how_to_guides/trigger_webhook), [How to Sync Prompts with GitHub](https://docs.smith.langchain.com/prompt_engineering/tutorials/prompt_commit)
+
+## How to use a prompt in production without repeated API calls
+
+In a production application, making a network request to fetch a prompt every time a function is executed adds unnecessary latency and creates a dependency that could fail. The application should be resilient and fast.
```

Comment:
```suggestion
In a production application, making a network request to fetch a prompt every time a function is executed adds unnecessary latency and creates a dependency that could fail.
```
What do you think about dropping this sentence, I think you cover this well in the next paragraph.